k8s/negotiator.go

```diff
@@ -98,3 +97,118 @@ func (*GenericJSONDecoder) Encode(obj runtime.Object, w io.Writer) error {
 func (*GenericJSONDecoder) Identifier() runtime.Identifier {
 	return "generic-json-decoder"
 }
+
+/*
```

Comment:
To start with, I have the client only work with JSON, as that's how it currently works (and SchemalessClient / ClientRegistry are explicit about requiring a a JSON codec). I'd like to make it work with other encodings in the future, but that ballooned a bit as I was working on that in the negotiator (primarily because of the complexity of the stream decoder), so for now I'm just maintaining current behavior.

Reply:
(basically, they need a `Framer`, which is a generic decoder from `[]byte` into `any`, which our `Codec` model doesn't provide. Need to figure out if we need to update the `Codec` or if we can deal with it in an alternative way).

Reply:
I can also remove this commented-out code, I kind of forgot I left it in there

Reply:
Okay, gotcha 👍